Add a demo of picking and moving object.

rongguodong opened the pull request:
Click to pick, click again to drop.
When an object is picked, it can be rotated with camera and moved along
radius by finger swipe.

GearVRf-DCO-1.0-Signed-off-by: Guodong Rong
rongguodong@hotmail.com

rongguodong commented:
This is a demo for VR Hackathon.

Since GVRMeshEyePointee does not work well with Future<GVRMesh>, I temporarily use a blocking mesh loading with FutureWrapper to make it working. When GVRMeshEyePointee is modified to be compatible with Future<GVRMesh>, we can make this demo better with asynchronous mesh loading.

thomasflynn commented:
ok, it runs and displays now.  though, it wasn't responding to touches on my screen.  didn't try in a headset yet.

thomasflynn commented:
ok, Guodong showed me how it works.  i just wasn't using it right.  Chao, could you do an initial review of the code?  thanks.

JonShemitz commented on GVRf/Framework/jni/gl/gl_texture.h:
```diff
@@ -32,7 +32,7 @@ class GLTexture {
         glTexParameteri(target, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
         glTexParameteri(target, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-        glTexParameteri(target, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+        glTexParameteri(target, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
```
This fixes the Solar System sample.

thomasflynn replied:
good.  though, why was i seeing a null mesh?  and why was it a random number of objects that were visible?

JonShemitz replied:
> why was i seeing a null mesh? and why was it a random number of objects that were visible?

That's a really good question. In all my poking around today, I never saw a `null` mesh or a `null` texture, whether with the original `.notify()` code or @guodongrong 's  `.notifyAll()` change.

JonShemitz commented on GVRf/Sample/gvr-pickandmove/AndroidManifest.xml:
```diff
@@ -0,0 +1,49 @@
+        android:minSdkVersion="16"
+        android:targetSdkVersion="19" />
+    
+    <uses-permission android:name="android.permission.MOUNT_UNMOUNT_FILESYSTEMS"/>
```
Not sure we need this (these?) permissions.

JonShemitz commented on GVRf/Sample/gvr-pickandmove/res/values-sw600dp/dimens.xml:
```diff
@@ -0,0 +1,25 @@
+         Customize dimensions originally defined in res/values/dimens.xml (such as
+         screen margins) for sw600dp devices (e.g. 7" tablets) here.
+    -->
+
```
Do we really need this empty dimensions file?

chenchao1407 commented:
OK. I am going to review it soon.

chenchao1407 commented on GVRf/Sample/gvr-pickandmove/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,4 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.6
+org.eclipse.jdt.core.compiler.compliance=1.6
+org.eclipse.jdt.core.compiler.source=1.6
```
I am not sure if it is necessary to merge.

JonShemitz replied:
Yeah, probably shouldn't.

chenchao1407 commented:
Code seems ok to me. Runs well both with and without Samsung Gear. Can be merged after Guodong removes the .settings folder.

rongguodong commented:
I have removed .setting and the unnecessary permissions. 

I also added head tracking indicator per Tom's request. Please review and test it again. Thanks!

rongguodong commented:
Rebased with top of tree.

chenchao1407 commented:
I think good to go with this. Any objection?

thomasflynn commented:
Runs, does what i expect.  Jon, i think he addressed your comments.  Any further comments?

JonShemitz commented:
>  Any further comments?

Nope.
